programs/stake/src/stake_state.rs

```diff
@@ -595,10 +595,15 @@ impl Stake {
         let rewards = u64::try_from(rewards).unwrap();
 
         // don't bother trying to split if fractional lamports got truncated
-        if rewards == 0 {
-            return None;
-        }
-        let (voter_rewards, staker_rewards, is_split) = vote_state.commission_split(rewards);
+        let (voter_rewards, staker_rewards, is_split) = if rewards == 0 {
```

Comment:
Would it be more accurate to use `point_value.rewards` in the conditional here? :thinking:

Reply:
well, I think this is same thing. :)

Reply:
I think the difference would be that `rewards` being `0` could result from either `points` being `0` or staking rewards being disabled (`points_value.rewards` being `0`).  We only want the new behavior in the latter case

Reply:
Ah! But the former has already been taken care of...
https://github.com/solana-labs/solana/blob/1ffab5de775b1772fdfd2b24e13e8594f8a63e25/programs/stake/src/stake_state.rs#L585-L587

/me mumbles obscenities about early returns